src/misc/lv_types.h

```diff
@@ -47,23 +31,10 @@ enum {
                        function or an operation was failed*/
     LV_RES_OK,      /*The object is valid (no deleted) after the action*/
 };
-typedef uint8_t lv_res_t;
 
-#if defined(__cplusplus) || __STDC_VERSION__ >= 199901L
-// If c99 or newer,  use the definition of uintptr_t directly from <stdint.h>
+typedef uint8_t lv_res_t;
 typedef uintptr_t lv_uintptr_t;
 
-#else
-
-// Otherwise, use the arch size determination
-#ifdef LV_ARCH_64
-typedef uint64_t lv_uintptr_t;
-#else
-typedef uint32_t lv_uintptr_t;
-#endif
```

Comment:
It was added this way because some compilers don't provide `uintptr_t`. E.g. Microchip's XC16,

Reply:
Ok, I keep uintptr_t logic, but remove the check around the inclusion of stdint.h.